app/presenters/publishing_api/role_presenter.rb

```diff
@@ -0,0 +1,72 @@
+module PublishingApi
+  class RolePresenter
+    attr_accessor :role, :update_type
+
+    def initialize(role, update_type: nil)
+      self.role = role
+      self.update_type = update_type || "major"
+    end
+
+    def content_id
+      role.content_id
+    end
+
+    def content
+      content = BaseItemPresenter.new(
+        role,
+        title: role.name,
+        update_type: update_type,
+      ).base_attributes
+
+      content.merge!(
+        description: nil,
+        details: details,
+        document_type: role.class.name.underscore,
+        links: {},
+        public_updated_at: role.updated_at,
+        rendering_app: Whitehall::RenderingApp::WHITEHALL_FRONTEND,
+        schema_name: schema_name,
+      )
+
+      content.merge!(PayloadBuilder::PolymorphicPath.for(role))
+    end
+
+  private
+
+    def schema_name
+      "role"
+    end
+
+    def details
+      {
+        permanent_secretary: role.permanent_secretary,
+        cabinet_member: role.cabinet_member,
+        slug: role.slug,
+        whip_organisation: whip_organisation,
+        whip_ordering: role.whip_ordering,
+        role_type: role.type.chomp("Role"),
+        seniority: role.seniority,
+        attends_cabinet_type: role.attends_cabinet_type,
+        role_payment_type: role.role_payment_type,
+        supports_historical_accounts: role.supports_historical_accounts,
+      }
+    end
+
+    def whip_organisation
```

Comment:
WhipOrganisations is not really a model

https://github.com/alphagov/whitehall/blob/master/lib/whitehall/whip_organisation.rb#L17-L21